src/coreclr/src/vm/jithelpers.cpp

```diff
@@ -5233,6 +5234,74 @@ void JIT_Patchpoint(int* counter, int ilOffset)
 
 #endif // FEATURE_ON_STACK_REPLACEMENT
 
+HCIMPL2(void, JIT_ClassProfile, Object *obj, void* tableAddress)
+{
+    FCALL_CONTRACT;
+    FC_GC_POLL_NOT_NEEDED();
+
+    OBJECTREF objRef = ObjectToOBJECTREF(obj);
+    VALIDATEOBJECTREF(objRef);
+
+    ICorJitInfo::ClassProfile* const classProfile = (ICorJitInfo::ClassProfile*) tableAddress;
+    volatile unsigned* pCount = (volatile unsigned*) &classProfile->Count;
+    const unsigned count = *pCount++;
+    const unsigned S = ICorJitInfo::ClassProfile::SIZE;
+    const unsigned N = ICorJitInfo::ClassProfile::SAMPLE_INTERVAL;
+
```

Comment:
assert(N >= S);  // SAMPLE_INTERVAL must be >= SIZE.

and possibly assert that N is a power of two

Reply:
N doesn't need to a power of 2, though it probably will be. in practice.